packages/connect/src/ws_orderbook_channel.ts

```diff
@@ -55,112 +49,84 @@ export class WebSocketOrderbookChannel implements OrderbookChannel {
-            });
-            this._client.connect(this._apiEndpointUrl);
+    /**
+     * Send a message to the client if it has been instantiated and it is open
+     */
+    private _sendMessage(message: any): void {
+        if (this._client.readyState === WebSocket.w3cwebsocket.OPEN) {
+            this._client.send(JSON.stringify(message));
         }
     }
-    private _handleWebSocketMessage(
-        requestId: number,
-        subscriptionOpts: OrderbookChannelSubscriptionOpts,
-        message: WebSocket.IMessage,
-        handler: OrderbookChannelHandler,
-    ): void {
-        if (!_.isUndefined(message.utf8Data)) {
-            try {
-                const utf8Data = message.utf8Data;
-                const parserResult = orderbookChannelMessageParser.parse(utf8Data);
-                if (parserResult.requestId === requestId) {
-                    switch (parserResult.type) {
-                        case OrderbookChannelMessageTypes.Snapshot: {
-                            handler.onSnapshot(this, subscriptionOpts, parserResult.payload);
-                            break;
-                        }
-                        case OrderbookChannelMessageTypes.Update: {
-                            handler.onUpdate(this, subscriptionOpts, parserResult.payload);
-                            break;
-                        }
-                        default: {
-                            handler.onError(
-                                this,
-                                subscriptionOpts,
-                                new Error(`Message has missing a type parameter: ${utf8Data}`),
-                            );
-                        }
-                    }
+    /**
+     * For use in cases where we need to alert all handlers of an error
+     */
+    private _alertAllHandlersToError(error: Error): void {
+        _.forEach(this._subscriptions, subscription => {
+            subscription.handler.onError(this, subscription.subscriptionOpts, error);
+        });
+    }
+    private _alertAllHandlersToClose(): void {
+        _.forEach(this._subscriptions, subscription => {
+            subscription.handler.onClose(this, subscription.subscriptionOpts);
+        });
+    }
+    private _handleWebSocketMessage(message: any): void {
+        // if we get a message with no data, alert all handlers and return
+        if (_.isUndefined(message.data)) {
+            this._alertAllHandlersToError(new Error(`Message does not contain utf8Data`));
+            return;
+        }
+        // try to parse the message data and route it to the correct handler
+        try {
+            const utf8Data = message.data;
+            const parserResult = orderbookChannelMessageParser.parse(utf8Data);
+            const subscription = this._subscriptions[parserResult.requestId];
+            if (_.isUndefined(subscription)) {
+                this._alertAllHandlersToError(new Error(`Message has unknown requestId: ${utf8Data}`));
```

Comment:
I think we should just console log this and drop move on, rather than alert every subscription. There's not much a subscription could do about it.

Reply:
But this is a developer error and should be fixed no? Throwing forces it to be fixed.